cnosdb/src/main/java/cn/edu/tsinghua/iot/benchmark/cnosdb/CnosDB.java

```diff
@@ -0,0 +1,488 @@
+      influxDbInstance = org.influxdb.InfluxDBFactory.connect(cnosUrl, client);
+      cnosConnection = new CnosConnection(cnosUrl, cnosDbName);
+    } catch (Exception e) {
+      LOGGER.error("Initialize CnosDB failed because ", e);
+      throw new TsdbException(e);
+    }
+  }
+
+  @Override
+  public void cleanup() throws TsdbException {
+    try {
+      Response response = cnosConnection.execute("DROP DATABASE IF EXISTS " + cnosDbName);
+      response.close();
+    } catch (Exception e) {
+      LOGGER.error("Cleanup CnosDB failed because ", e);
+      throw new TsdbException(e);
+    }
+  }
+
+  @Override
+  public void close() {
+    if (influxDbInstance != null) {
+      influxDbInstance.close();
+    }
+  }
+
+  @Override
+  public Double registerSchema(List<DeviceSchema> schemaList) throws TsdbException {
+    long start;
+    long end;
+    try {
+      start = System.nanoTime();
+      Response response =
+          cnosConnection.execute(
+              String.format(
+                  "create database if not exists %s with shard %d",
+                  cnosDbName, config.getCNOSDB_SHARD_NUMBER()));
+      response.close();
+      end = System.nanoTime();
+    } catch (Exception e) {
+      LOGGER.error("RegisterSchema CnosDB failed because" + e.getMessage());
+      throw new TsdbException(e);
+    }
+    return TimeUtils.convertToSeconds(end - start, "ns");
+  }
+
+  @Override
+  public Status insertOneBatch(IBatch batch) {
+    BatchPoints batchPoints = BatchPoints.database(cnosDbName).build();
+    try {
+      InfluxDataModel model;
+      for (Record record : batch.getRecords()) {
+        model =
+            createDataModel(
+                batch.getDeviceSchema(), record.getTimestamp(), record.getRecordDataValue());
+        batchPoints.point(model.toInfluxPoint());
+      }
+
+      influxDbInstance.write(batchPoints);
+
+      return new Status(true);
+    } catch (Exception e) {
+      LOGGER.warn(e.getMessage());
+      return new Status(false, 0, e, e.toString());
+    }
+  }
+
+  private InfluxDataModel createDataModel(
+      DeviceSchema deviceSchema, Long time, List<Object> valueList) throws TsdbException {
+    InfluxDataModel model = new InfluxDataModel();
+    model.setMeasurement(deviceSchema.getGroup());
+    HashMap<String, String> tags = new HashMap<>();
+    tags.put("device", deviceSchema.getDevice());
+    model.setTagSet(tags);
+    model.setTimestamp(time);
+    model.setTimestampPrecision(config.getTIMESTAMP_PRECISION());
+    HashMap<String, Object> fields = new HashMap<>();
+    List<Sensor> sensors = deviceSchema.getSensors();
+    for (int i = 0; i < sensors.size(); i++) {
+      fields.put(sensors.get(i).getName(), valueList.get(i));
+    }
+    model.setFields(fields);
+    return model;
+  }
+
+  /** eg. SELECT s_0 FROM group_2 WHERE ( device = 'd_8' ) AND time = 1535558405000000000. */
+  @Override
+  public Status preciseQuery(PreciseQuery preciseQuery) {
```

Comment:
If these queries are exactly the same as Influxdb's implementation, is it necessary to override them?